src/Nethermind/Nethermind.Consensus/Processing/ReadOnlyTxProcessingScope.cs

```diff
@@ -1,28 +1,23 @@
 // SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
-using Nethermind.Core.Crypto;
+using System;
 using Nethermind.Evm.TransactionProcessing;
 using Nethermind.State;
 
 namespace Nethermind.Consensus.Processing;
 
 public class ReadOnlyTxProcessingScope(
     ITransactionProcessor transactionProcessor,
-    IWorldState worldState,
-    Hash256 originalStateRoot
+    IWorldState worldState
 ) : IReadOnlyTxProcessingScope
 {
     public void Dispose()
     {
-        Reset();
+        worldState.Reset();
+        worldState.SetBaseBlock(null);
```

Comment:
No more original state root. It is always reset to empty state. Consumer should always set base block first.